feat: Default `is_ai_data_processing_approved` to True

rafaeelaudibert opened the pull request:
We believe that by defaulting this to True our users will be able to extract more value from PostHog. It's already easy to enable AI but it's restricted to admins, which hurts the ability for non-admin people to start using it.

People can still opt-out if they want, that's not been removed and never will. We aren't training on this data either.

rafaeelaudibert committed:
feat: Default `is_ai_data_processing_approved` to True

We believe that by defaulting this to True our users will be able to extract more value from PostHog. It's already easy to enable AI but it's restricted to admins, which hurts the ability for non-admin people to start using it.

People can still opt-out if they want, that's not been removed and never will. We aren't training on this data either.

rafaeelaudibert commented:
* **#43955** <a href="https://app.graphite.com/github/pr/PostHog/posthog/43955?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/PostHog/posthog/43955?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

github-actions[bot] committed:
test(storybook): update UI snapshots

Twixes commented on frontend/src/scenes/onboarding/OnboardingAIConsent.tsx:
```diff
@@ -27,7 +27,11 @@ export const OnboardingAIConsent = ({ stepKey }: { stepKey: OnboardingStepKey })
-    const [aiEnabled, setAiEnabled] = useState(currentOrganization?.is_ai_data_processing_approved ?? false)
+
+    // New accounts have `default=True` on the database, but older accounts may have null/false
+    const [isEnabledByDefault] = useState(!!currentOrganization?.is_ai_data_processing_approved)
```
Why is this useState, rather than just a plain const? There's certainly more overhead in the useState here than worth for the boolean computation

rafaeelaudibert replied:
Ah, the idea here was to be able to store the original value for the setting without the changes the user might've taken in this screen. I now see we actually don't store the changes to `organization`until we move to the next step, so this isn't needed

rafaeelaudibert committed:
refactor: Stop using useless `useState`

The idea here was to be able to store the original value for the setting without the changes the user might've taken in this screen. I now see we actually don't store the changes to organizationuntil we move to the next step, so this isn't needed

github-actions[bot] committed:
test(storybook): update UI snapshots
